src/Chapter2.hs

```diff
@@ -898,8 +901,11 @@ and reverses it.
   cheating!
 -}
 rewind :: [a] -> [a]
-rewind [] = []
-rewind x = last x : rewind (init x)
+rewind x = go [] x
+  where
+    go :: [a] -> [a] -> [a]
+    go l [] = l
+    go l (x:xs) = go (x:l) xs
```

Comment:
Would this be a valid approach to implement reverse? 
Also, while re-working this I came across `foldl` and would like to know more about it 🤔 . Could you recommend some good resources/examples?

Reply:
Yes, this is absolutely right! 💯 

As you correctly noticed, `foldl` can be used here. It is the method of the `Foldable` typeclass. Typeclasses are covered in the following Chapter of this course, but if you are interested to learn more now I can recommend this amazing blog post: https://www.aymannadeem.com/programming/languages/2019/04/24/Foldilocks.html